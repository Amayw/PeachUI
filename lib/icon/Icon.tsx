import React from 'react';
import './icon.scss';
// import './svg'
import classes from '../helpers/classes';

interface IconProps extends React.SVGAttributes<SVGElement>{
    name:string
}

const Icon:React.FC<IconProps>=({className,name,...restProps})=>{
    return (
            <svg className={classes('icon', className)} {...restProps}>
                <use xlinkHref={`#i-${name}`}/>
            </svg>
    )
}

export default Icon;
