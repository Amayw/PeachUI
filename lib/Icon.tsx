import React from 'react';
import './svg'
import './icon.scss'
interface IconProps {
    name:string
}
const Icon:React.FC<IconProps>=(props)=>{
    return (
            <svg className='icon'>
                <use xlinkHref={`#${props.name}`} />
            </svg>
    )
}

export default Icon;
