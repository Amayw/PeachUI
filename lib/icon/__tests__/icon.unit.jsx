import renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../Icon'
import {mount} from 'enzyme'

describe("icon",()=>{
    it('可以接收name',()=>{
        const json=renderer.create(<Icon name='left'/>).toJSON();
        expect(json).toMatchSnapshot();
    })

    it('可以被点击',()=>{
        const fn=jest.fn();
        const component=mount(<Icon name='left' onClick={fn}/>);
        component.find('svg').simulate('click');
        expect(fn).toBeCalled();
    })
})
