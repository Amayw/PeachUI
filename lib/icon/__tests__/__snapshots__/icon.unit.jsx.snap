// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 可以接收name 1`] = `
<svg
  className="icon"
>
  <use
    xlinkHref="#i-left"
  />
</svg>
`;
